fda/basis.py

```diff
@@ -1778,6 +1784,21 @@ def copy(self, *, basis=None, coefficients=None, dataset_label=None,
                           axes_labels=axes_labels, extrapolation=extrapolation,
                           keepdims=keepdims)
 
+    def inprod(self, other):
+        if self.domain_range != other.domain_range:
```

Comment:
After the creation of `FData` the `domain_range` is stored in an array with the domain ranges of each dimension, to treat `FDataBasis` and  `FDataGrid` uniformly.

Reply:
I've change it to only see to the first item as is used in

https://github.com/GAA-UAM/fda/blob/92c984cb42a505661716c26f4cd56a2f1e69e26f/fda/basis.py#L218

It will be updated in next steps of the development